Automattic/jetpack
# Jetpack AI: Add machinery to generate audio transcriptions

Fixes #35548.

## Proposed changes:

* Created a `transcribeAudio` helper function on the `ai-client` package to call the transcription endpoint
* Created an `useAudioTranscription` hook on the `ai-client` package to abstract details from the helper function and simpler usage on React
* Created a simple, throwaway component on the current Voice-To-Content block implementation to test/demo the hook usage

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor
* Open the browser console to watch the debug messages (make sure to enable it with `localStorage.setItem( 'debug', '*' )`
* Insert a "Voice-to-Content" block
* You will notice a button to pick an audio file for transcription:

<img width="600" alt="Screenshot 2024-02-15 at 14 41 53" src="https://github.com/Automattic/jetpack/assets/6760046/5198df0b-4e36-4dc6-a180-23f02f4ef401">

* Click it and choose a short audio file (tip: download an OpenAI demo file at https://cdn.openai.com/whisper/draft-20220913a/micro-machines.wav)
* Once you choose the file, you will see some debug messages on the console, related to fetching a JWT token as well as requesting the transcription:

<img width="600" alt="Screenshot 2024-02-14 at 16 02 46" src="https://github.com/Automattic/jetpack/assets/6760046/3a455cc2-b4da-42eb-a533-0f0bc2b15710">

* When the transcription is done, you will see the content of it on the console as well:

<img width="600" alt="Screenshot 2024-02-15 at 14 41 44" src="https://github.com/Automattic/jetpack/assets/6760046/458277d5-1679-467b-8848-90988124857d">

* Remember this flow is just a way to **test it**, the important thing is the internal flow to get a token and request the transcription, as well as handling the events from it
* We will need to adjust the hook usage once #35547 is ready
* **Known issue:** the backend is refusing `mp3` files because the is `audio/mpeg`, and this type is not on the list of allowed types (it's allowing `audio/mp3` instead); D138285-code will fix it
   * Tip: use a `mp3` file to test the error handling, to see an error message on the console:

<img width="600" alt="Screenshot 2024-02-15 at 14 53 22" src="https://github.com/Automattic/jetpack/assets/6760046/06e6a853-ca06-430d-94b2-801d49828fd0">
